tests/backup/backup_cluster_share_test.go

```diff
@@ -0,0 +1,481 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/pborman/uuid"
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+)
+
+var _ = Describe("{ClusterShare}", Label(TestCaseLabelsMap[ClusterShare]...), func() {
+	type user struct {
+		name string
+		ctx  context.Context
+	}
+
+	var (
+		scheduledAppContexts []*scheduler.Context
+		bkpNamespaces        []string
+		providers            []string            = GetBackupProviders()
+		infraAdminRole       backup.PxBackupRole = backup.InfrastructureOwner
+		superAdminRole       backup.PxBackupRole = backup.SuperAdmin
+		cloudCredName        string
+		cloudCredUID         string
+		backupLocationUID    string
+		bkpLocationName      string
+		backupLocationMap    map[string]string
+		testUsers            map[int]user
+	)
+
+	JustBeforeEach(func() {
+		backupLocationMap = make(map[string]string)
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		bkpNamespaces = make([]string, 0)
+		cloudCredName, cloudCredUID, backupLocationUID, bkpLocationName = "", "", "", ""
+
+		// Schedule an Application
+		appContexts := ScheduleApplications(TaskNamePrefix)
+		for _, ctx := range appContexts {
+			ctx.ReadinessTimeout = AppReadinessTimeout
+			namespace := GetAppNamespace(ctx, TaskNamePrefix)
+			bkpNamespaces = append(bkpNamespaces, namespace)
+			scheduledAppContexts = append(scheduledAppContexts, ctx)
+		}
+
+		// Validate the application
+		ValidateApplications(scheduledAppContexts)
+
+		// Create two test users with infra admin role
+		testUsers = map[int]user{1: {name: "testuser1"}, 2: {name: "testuser2"}}
+		log.InfoD("Creating %d users", 2)
+		for i := 1; i <= 2; i++ {
+			userName := testUsers[i].name
+			firstName := fmt.Sprintf("FirstName%v", i)
+			lastName := fmt.Sprintf("LastName%v", i)
+			email := fmt.Sprintf("testuser%v@cnbu.com", i)
+
+			err := backup.AddUser(userName, firstName, lastName, email, CommonPassword)
+			log.FailOnError(err, "Failed to create user - %s", userName)
+
+			err = backup.AddRoleToUser(userName, infraAdminRole, fmt.Sprintf("Adding %v role to %s", infraAdminRole, userName))
+			log.FailOnError(err, "failed to add role %s to the user %s", infraAdminRole, userName)
+		}
+
+	})
+
+	// This testcase verifies whether the restores created/owned by the user were deleted during the cluster un-share.
+	It("VerifyRestoreObjectsAreDeletedCreatedByNonSuperAdmin", func() {
+		//TODO: Need to update the testrail ID
+		StartPxBackupTorpedoTest("VerifyRestoreObjectsAreDeletedCreatedByNonSuperAdmin", "VerifyRestoreObjectsAreDeletedCreatedByNonSuperAdmin during the cluster unshare", nil, 0, Sgajawada, Q2FY25)
```

Comment:
Please make and entry of your name in https://github.com/portworx/torpedo/blob/master/tests/backup_helper.go#L80

Reply:
Also, please do not add any automation test without importing it into testrail. Testrail ID is mandatory